src/index.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-alert */
+import genHTML from './gen_html.js';
+import addNew from './add.js';
+import './style.css';
+
+let listArr = [];
+
+const list = document.getElementById('list');
+
+const reload = () => {
+  if (localStorage.getItem('todoList')) {
+    const oldStorage = localStorage.getItem('todoList');
+    const newStorage = JSON.parse(oldStorage);
+    list.textContent = '';
+    listArr = newStorage;
+    genHTML(list, listArr);
+  } else {
+    const defaultItem = [
+      {
+        description: 'Add your first task',
+        completed: false,
+        index: 1,
+      },
+    ];
+    const newStorage = JSON.stringify(defaultItem);
+    listArr = defaultItem;
+    localStorage.setItem('todoList', newStorage);
+    list.innerHTML = '';
+    genHTML(list, JSON.parse(newStorage));
+  }
+  const checkboxes = document.querySelectorAll('.checkbox');
+  for (let i = 0; i < checkboxes.length; i += 1) {
+    checkboxes[i].addEventListener('change', () => {
+      reload();
+    });
+  }
+};
+
+window.onload = reload();
+
+const task = document.getElementById('add-item');
+const addNewBtn = document.getElementById('add-new-btn');
+addNewBtn.addEventListener('click', () => {
+  if (task.value.length > 0) {
+    listArr.push(addNew(task.value, listArr.length + 1));
+    const newStorage = JSON.stringify(listArr);
+    localStorage.setItem('todoList', newStorage);
+    task.value = '';
+    reload();
+  } else {
+    alert('Tasks must contain text.\nTry typing something into the "Add to your list..." input.');
```

Comment:
- [ ][Optional] It might be nicer to have a custom modal instead of an alert. Maybe you can have a separate module for handling alert-like modals which you can add to your later projects. There are also some packages for modals.
As a reminder, alerts stop code execution along with other concerns.

Reply:
Thank you ! Could you show a example from modal?

Reply:
Thank you for your patience. Although I didn't notice your comment until recently.
You might have been able to find some inspiration on that. Nevertheless, you can check this [example](https://codepen.io/colbyalbo/pen/gRogbE)
also this [video](https://www.youtube.com/watch?v=iE_6pQ3RlZU&pp=ygUMc2ltcGxlIG1vZGFs)